### DISCUSSION TITLE
Wix v4.preview.1 EXE/CLI not picking up extensions

### DISCUSSION THREAD

**Discussion by 0xspeedybird** [wixtoolset/issues#7073] [Questions]
I've tried to include the bal ext to build a burn bundle but no matter the arguments I supply, the build fails with the below error.  Which syntax should work to create a bundle? Any pointers as to what is wrong?

`error WIX0200: The BootstrapperApplication element contains an unhandled extension element 'WixStandardBootstrapperApplication'. Please ensure that the extension for elements in the 'http://wixtoolset.org/schemas/v4/wxs/bal' namespace has been provided.`

My simple bundle (ignore space after opening tag, thats there due to Github but not in code):

< Wix xmlns="http://wixtoolset.org/schemas/v4/wxs" xmlns:bal="http://wixtoolset.org/schemas/v4/wxs/bal">

    <Bundle Name="WixStdBa" Version="1.0.0.0" Manufacturer="My Co" UpgradeCode="75D5D534-E177-4689-AAE9-CAC1C39002C2">

        <BootstrapperApplication>
            <bal:WixStandardBootstrapperApplication LicenseUrl="http://wixtoolset.org/about/license/" Theme="hyperlinkLicense" />
        </BootstrapperApplication>

        <Chain>
            <ExePackage Permanent="yes" DetectCondition="none" SourceFile="$(var.ProjectDir)\myapp.exe" />
        </Chain>

    </Bundle>
</Wix>

Set up (run via Github actions after installing .NET:

`dotnet tool install --global wix --version 4.0.0-preview.1 

wix extension add WixToolset.UI.wixext

wix extension add WixToolset.Util.wixext

wix extension add WixToolset.Bal.wixext`

Various attempts that all failed (notice the difference in the ext reference with the -ext flag):

`wix build -src "$env:GITHUB_WORKSPACE\msi_build\Bundle.wxs" -d var.ProjectDir=$env:GITHUB_WORKSPACE\msi_build -ext WixToolset.Bal.wixext -o releases\SimpleBundle.msi`

And -d variations:

`-ext WixToolset.Bal.wixext.dll`
and:

`-ext $HOME\.nuget\packages\wixtoolset.bal.wixext\4.0.0-preview.1\wixext4\WixToolset.Bal.wixext.dll`

When I last ran it today with "wix extension list", I got this which seemed noteworthy:

`WixToolset.Bal.wixext 4.0.0-preview.1 (damaged)

WixToolset.UI.wixext 4.0.0-preview.1-build.590 (damaged)

WixToolset.Util.wixext 4.0.0-preview.1-build.590 (damaged)`

Please note that I am able to build MSIs with the UI extensions in another process.

**Comment by 0xspeedybird** [comment]
Here is a reference with the full path to the extension DLL being used.  I am hoping to understand what is the correct syntax.  Thanks.

`wix build -ext C:\my-project\.wix\extensions\WixToolset.UI.wixext\4.0.0-preview.0\tools\WixToolset.UI.wixext.dll installer.wxs`

https://github.com/wixtoolset/issues/discussions/6516#discussioncomment-1030590

which led to this issue being resolved:

https://github.com/wixtoolset/issues/issues/6536

**Reply by robmen** [reply]
It looks like you are trying to use a `-preview.0` extension with `-preview.1` executable. That definitely won't work. I'm a bit confused about how you got into a state that has both `-preview.0` and `-preview.1` extensions installed.

**Reply by 0xspeedybird** [reply]
That is likely a typo.  Is this implying that I must provide the full path to the extension DLL and cannot use the same syntax the "wix extension add" command supports?  I ask as it look like 6536 is supposed to resolve that.

**Reply by robmen** [reply]
> Is this implying that I must provide the full path to the extension DLL

No.

One of the internal directory paths changed between `-preview.0` to `-preview.1`. That full path you provided is definitely the old `-preview.0` path (that'd need to be multiple typos). So I was drilling into that piece of information because it didn't align with the other information you provided.

Basically, I'm trying to discern the exact minimal steps needed to reproduce the issue before spending time digging in.

**Reply by 0xspeedybird** [reply]
Sorry, I was trying to show evidence of trying out various values for the ext flag.  This is the current set of commands in the build.  It fails with the error cited above after the last line.  Does that help clarify?

dotnet tool install --global wix --version 4.0.0-preview.1

wix extension add WixToolset.UI.wixext

wix extension add WixToolset.Util.wixext

wix extension add WixToolset.Bal.wixext

wix build -src "$env:GITHUB_WORKSPACE\msi_build\Bundle.wxs" -d var.ProjectDir=$env:GITHUB_WORKSPACE\msi_build -ext WixToolset.Bal.wixext -o releases\SimpleBundle.msi

**Reply by robmen** [reply]
Yes. Unfortunately, I follow those steps, and I'm unable to reproduce the build issue. Can you provide a link to a little GitHub repository with a batch file or something that can reproduce the issue?

I can reproduce the `(damaged)` reports from `wix extension list`. That's a bug.

**Reply by 0xspeedybird** [reply]
I moved the above example to a simple project and it worked.  Good call.  When it was combined with my other MSI Wix project, it would not work.  I have decided to refactor and more strictly isolated my MSI and Bundle WXS files via subfolders.  It's unclear what is the root cause as of yet.  Thanks again for the suggestion.

**Reply by robmen** [reply]
If you can provide a repro of the issue and share it in a GitHub repo, it'd be interesting to track down the root issue.

The erroneous "(damaged)" message from `wix extension list` was #7084 and is fixed in -rc.1.

**Reply by 0xspeedybird** [reply]
Thanks for the offer.  Unfortunately, the project is closed source and not easily extracted into a reproducible sample.  What resolved it for me was moving my bundle WXS into another directory so it wasn't comingled with my MSI build.  After that, I experienced no more issues.  I subsequently managed to build a DLL with my own custom action.  This has eliminated the need for the bundle and bal extension.
